### ISSUE TITLE
Einsum fails on certain configurations for CuArrays

### ISSUE THREAD

**Issue by vilim** [under-Peter/OMEinsum.jl#78]
This is a wonderful package and gives great performance on CPU.
However, while GPU support works for some cases, it fails on others:

```
a = CuArrays.ones((10,10))
b = CuArrays.ones((5,10))
c = CuArrays.ones((5,10))

res = ein"ij,li,mi->lm"(a,b,c) # works
res = ein"ij,li,mj->lm"(a,b,c) # fails (last i changed to j)
```
the error is 

```
MethodError: no method matching Strided.UnsafeStridedView(::CUDAdrv.CuPtr{Float32}, ::Tuple{Int64,Int64}, ::Tuple{Int64,Int64})
Closest candidates are:
  Strided.UnsafeStridedView(!Matched::Ptr{T}, ::Tuple{Vararg{Int64,N}}, ::Tuple{Vararg{Int64,N}}) where {T, N} at C:\Users\portugueslab\.julia\packages\Strided\qA9SM\src\unsafestridedview.jl:18
  Strided.UnsafeStridedView(!Matched::Ptr{T}, ::Tuple{Vararg{Int64,N}}, ::Tuple{Vararg{Int64,N}}, !Matched::Int64) where {T, N} at C:\Users\portugueslab\.julia\packages\Strided\qA9SM\src\unsafestridedview.jl:18
  Strided.UnsafeStridedView(!Matched::Ptr{PT}, ::Tuple{Vararg{Int64,N}}, ::Tuple{Vararg{Int64,N}}, !Matched::Int64, !Matched::F) where {PT, N, F} at C:\Users\portugueslab\.julia\packages\Strided\qA9SM\src\unsafestridedview.jl:12

Stacktrace:
 [1] Strided.UnsafeStridedView(::CuArray{Float32,2,Nothing}) at C:\Users\portugueslab\.julia\packages\Strided\qA9SM\src\unsafestridedview.jl:19
 [2] macro expansion at .\gcutils.jl:87 [inlined]
 [3] isblascontractable(::CuArray{Float32,2,Nothing}, ::Tuple{Int64}, ::Tuple{Int64}, ::Symbol) at C:\Users\portugueslab\.julia\packages\TensorOperations\aTVdS\src\implementation\stridedarray.jl:321
 [4] contract!(::Bool, ::CuArray{Float32,2,Nothing}, ::Symbol, ::CuArray{Float32,2,Nothing}, ::Symbol, ::Bool, ::CuArray{Float32,2,Nothing}, ::Tuple{Int64}, ::Tuple{Int64}, ::Tuple{Int64}, ::Tuple{Int64}, ::Tuple{Int64,Int64}, ::Tuple{Symbol,Symbol,Symbol}) at C:\Users\portugueslab\.julia\packages\TensorOperations\aTVdS\src\implementation\stridedarray.jl:265
 [5] contract! at C:\Users\portugueslab\.julia\packages\TensorOperations\aTVdS\src\implementation\stridedarray.jl:85 [inlined]
 [6] macro expansion at C:\Users\portugueslab\.julia\packages\OMEinsum\Yd4RG\src\einsum.jl:53 [inlined]
 [7] einsum at C:\Users\portugueslab\.julia\packages\OMEinsum\Yd4RG\src\einsum.jl:53 [inlined]
 [8] macro expansion at C:\Users\portugueslab\.julia\packages\OMEinsum\Yd4RG\src\einsum.jl:32 [inlined]
 [9] einsum(::EinCode{(('i', 'j'), ('l', 'i'), ('m', 'j')),('l', 'm')}, ::Tuple{CuArray{Float32,2,Nothing},CuArray{Float32,2,Nothing},CuArray{Float32,2,Nothing}}, ::IndexSize{6,Char}) at C:\Users\portugueslab\.julia\packages\OMEinsum\Yd4RG\src\einsum.jl:32
 [10] #call#47(::Nothing, ::EinCode{(('i', 'j'), ('l', 'i'), ('m', 'j')),('l', 'm')}, ::CuArray{Float32,2,Nothing}, ::Vararg{CuArray{Float32,2,Nothing},N} where N) at C:\Users\portugueslab\.julia\packages\OMEinsum\Yd4RG\src\interfaces.jl:41
 [11] (::EinCode{(('i', 'j'), ('l', 'i'), ('m', 'j')),('l', 'm')})(::CuArray{Float32,2,Nothing}, ::Vararg{CuArray{Float32,2,Nothing},N} where N) at C:\Users\portugueslab\.julia\packages\OMEinsum\Yd4RG\src\interfaces.jl:37
 [12] top-level scope at In[59]:1
```
Packages are at versions:
OMEinsum 0.3.0
CuArrays 1.4.1
with Julia 1.2 on Windows.

**Comment by GiggleLiu** [comment]
Thanks very much for your report, will fix it. You may add a pair of parentheses to circumvent ‘res = ein"(ij,li),mj->lm"(a,b,c)’ for the moment.

**Comment by GiggleLiu** [comment]
should be fixed by #80

**Comment by GiggleLiu** [comment]
Should be fixed by #80

```julia
julia> a = CuArrays.ones((10,10))
10×10 CuArray{Float32,2,Nothing}:
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0

julia> b = CuArrays.ones((5,10))
5×10 CuArray{Float32,2,Nothing}:
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0

julia> c = CuArrays.ones((5,10))
5×10 CuArray{Float32,2,Nothing}:
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0

julia> res = ein"ij,li,mi->lm"(a,b,c) # works
[ Info: Building the CUDAnative run-time library for your sm_50 device, this might take a while...
5×5 CuArray{Float32,2,CuArray{Float32,4,CuArray{Float32,2,Nothing}}}:
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0

julia> res = ein"ij,li,mj->lm"(a,b,c) # fails (last i changed to j)
5×5 CuArray{Float32,2,CuArray{Float32,3,Nothing}}:
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
 100.0  100.0  100.0  100.0  100.0
```
